test/scheduler_test.rb

```diff
@@ -456,6 +456,52 @@
            /#{Resque::Scheduler.send(:internal_name)}: cage/
   end
 
+  test 'gracefully shuts down rufus-scheduler threads' do
+    class BeforeEnqueueJob
+      @queue = :quick
+
+      def self.before_enqueue_example(*)
+        if Resque.redis.setnx('before_enqueue_job:enqueued', 'true')
+          sleep 5 # sleep longer than test timeout
```

Comment:
could use a different signal (and send both) but I guess that will become a bit messy